lib/mayaUsd/python/wrapConverter.cpp

```diff
@@ -0,0 +1,155 @@
+
+    return false;
+}
+
+static void convertMPlugToUsdAttr(
+    const Converter&     self,
+    const std::string&   attrName,
+    UsdAttribute&        usdAttr,
+    const ConverterArgs& args)
+{
+    MPlug plug;
+    if (UsdMayaUtil::GetPlugByName(attrName, plug) == MS::kSuccess)
+        self.convert(plug, usdAttr, args);
+}
+
+static void convertUsdAttrToMPlug(
+    const Converter&     self,
+    const UsdAttribute&  usdAttr,
+    const std::string&   attrName,
+    const ConverterArgs& args)
+{
+    MPlug plug;
+    if (UsdMayaUtil::GetPlugByName(attrName, plug) == MS::kSuccess)
+        self.convert(usdAttr, plug, args);
+}
+
+static VtValue
+convertMPlugToVtValue(const Converter& self, const std::string& attrName, const ConverterArgs& args)
+{
+    VtValue value;
+    MPlug   plug;
+    if (UsdMayaUtil::GetPlugByName(attrName, plug) == MS::kSuccess)
+        self.convert(plug, value, args);
+
+    return value;
+}
+
+static void convertVtValueToMPlug(
+    const Converter&     self,
+    const VtValue&       value,
+    const std::string&   attrName,
+    const ConverterArgs& args)
+{
+    MPlug plug;
+    if (UsdMayaUtil::GetPlugByName(attrName, plug) == MS::kSuccess)
+        self.convert(value, plug, args);
+}
+
+static void test_convertUsdAttrToMDGModifier(
+    const Converter&     self,
+    const UsdAttribute&  usdAttr,
+    const std::string&   attrName,
+    const ConverterArgs& args)
+{
+    MPlug plug;
+    if (UsdMayaUtil::GetPlugByName(attrName, plug) == MS::kSuccess) {
+        MDGModifier modifier;
+        self.convert(usdAttr, plug, modifier, args);
+
+        modifier.doIt();
+    }
+}
+
+static void test_convertVtValueToMDGModifier(
+    const Converter&     self,
+    const VtValue&       value,
+    const std::string&   attrName,
+    const ConverterArgs& args)
+{
+    MPlug plug;
+    if (UsdMayaUtil::GetPlugByName(attrName, plug) == MS::kSuccess) {
+        MDGModifier modifier;
+        self.convert(value, plug, modifier, args);
+
+        modifier.doIt();
+    }
+}
+} // namespace
+
+void wrapConverterArgs()
+{
+    using This = ConverterArgs;
+    class_<This>("ConverterArgs")
+        .def_readwrite("timeCode", &This::_timeCode)
+        .def_readwrite("doGammaCorrection", &This::_doGammaCorrection);
+}
+
+void wrapConverter()
+{
+    using This = Converter;
+    class_<This>("Converter", no_init)
+        .def("find", find1, return_value_policy<reference_existing_object>())
+        .def("find", find2, return_value_policy<reference_existing_object>())
+        .staticmethod("find")
+        .def("validate", validate)
+        .def("convert", convertMPlugToUsdAttr)
+        .def("convert", convertUsdAttrToMPlug)
+        .def("convertVt", convertMPlugToVtValue)
+        .def("convertVt", convertVtValueToMPlug)
+        .def("test_convertAndSetWithModifier", test_convertUsdAttrToMDGModifier)
+        .def("test_convertVtAndSetWithModifier", test_convertVtValueToMDGModifier);
```

Comment:
I agree with you, this will be a nice improvement. Funny enough I was inspired by code from VtValue tests. I will consider making this change as a separate PR.